### ISSUE TITLE
[AVM Question/Feedback]: Peerings fail with a missing feature, but the feature appears registered

### ISSUE THREAD

**Issue by SvenAelterman** [Azure/terraform-azurerm-avm-res-network-virtualnetwork#111]
### Check for previous/existing GitHub issues

- [X] I have checked for previous/existing GitHub issues

### Description

When specifying `peerings`, I am getting an error that the `Microsoft.Network/AllowMultiplePeeringLinksBetweenVnets` feature isn't registered.

However, when I run `Get-AzProviderFeature -ProviderNamespace Microsoft.Network -FeatureName AllowMultiplePeeringLinksBetweenVnets`, the state shows as `Registered` for the subscription where the error applies.

What can I do to resolve this issue?

**Comment by SvenAelterman** [comment]
OK, interesting. I checked on the remote side and the feature was not registered there. Even though the Terraform error clearly indicated that the subscription ID where it said the feature wasn't registered was the local side.

I went ahead and registered the feature on the remote side and then re-ran Terraform.

I am no longer getting that error, but now there's a different error, on both peerings:

```
│ --------------------------------------------------------------------------------
│ RESPONSE 400: 400 Bad Request
│ ERROR CODE: LocalAndRemoteSpaceMustContainEitherIpv4orIpv6Prefix
│ --------------------------------------------------------------------------------
│ {
│   "error": {
│     "code": "LocalAndRemoteSpaceMustContainEitherIpv4orIpv6Prefix",
│     "message": "Cannot create or update peering /subscriptions/fca9eaf6-728a-4252-9e2d-aa2532c82614/resourceGroups/network-demo-rg-cnc-01/providers/Microsoft.Network/virtualNetworks/hub-demo-vnet-cnc-01/virtualNetworkPeerings/hub-to-identity. Both Local and Remote Address Space in a Virtual Network Peering must contain either atleast one IPv4 in both or IPv6 prefix in both. Peered Spaces are, Local Address Space:  and Remote Address Space: ",
│     "details": []
│   }
│ }
│ --------------------------------------------------------------------------------
```

(I have also tried manually creating the peerings and that works.)

**Comment by jaredfholgate** [comment]
Let me check on this, we added a capability to register features for preview features. All our examples work successfully, so would be useful to see your Terraform code to be able to reproduce.

**Comment by Arhughes14** [comment]
@SvenAelterman ERROR CODE: LocalAndRemoteSpaceMustContainEitherIpv4orIpv6Prefix. How did you resolve this error?

**Comment by jaredfholgate** [comment]
> @SvenAelterman ERROR CODE: LocalAndRemoteSpaceMustContainEitherIpv4orIpv6Prefix. How did you resolve this error?

This should have been fixed in v0.4.0
